strkey.go

```diff
@@ -289,3 +291,55 @@ func (p PrefixByte) String() string {
 	}
 	return "unknown"
 }
+
+// Kind returns the PrefixByte for the role of the KeyPair
+// That is Operator, Account, User, Cluster, Server or Unknown
+func KeyPairRole(kp KeyPair) (PrefixByte, error) {
+	pk, err := kp.PublicKey()
+	if err != nil {
+		return PrefixByteUnknown, err
+	}
+
+	if IsValidPublicOperatorKey(pk) {
+		return PrefixByteOperator, nil
+	}
+
+	if IsValidPublicAccountKey(pk) {
+		return PrefixByteAccount, nil
+	}
+
+	if IsValidPublicUserKey(pk) {
+		return PrefixByteUser, nil
+	}
+
+	if IsValidPublicServerKey(pk) {
+		return PrefixByteServer, nil
+	}
+
+	if IsValidPublicClusterKey(pk) {
+		return PrefixByteCluster, nil
+	}
+
+	return PrefixByteUnknown, nil
+}
+
+// ValidateKeyPairRole returns an error if the KeyPair doesn't have an expected PrefixByte
+func ValidateKeyPairRole(kp KeyPair, expected ...PrefixByte) error {
+	pkType, err := KeyPairRole(kp)
+	if err != nil {
+		return err
+	}
+
+	for _, k := range expected {
+		if pkType == k {
+			return nil
+		}
+	}
+
+	var a []string
```

Comment:
Can just do fixed wrong type error. They have the info to debug IMO.

Reply:
I understand, but the caller has what they expect and what they are comparing it to, so when they get an error they are either debugging or its in prod and they will not parse the error IMO.

Reply:
I changed it to be the static error

Reply:
We can leave as is, just was trying to reason where it would be useful, and IMO only debugging.